### DISCUSSION TITLE
Differences between IronPython 2.7 and Pythonnet 3.0

### DISCUSSION THREAD

**Discussion by kcmparrot** [pythonnet/pythonnet#2193] [Q&A]
I have a project that is currently executing using IronPython 2.7  and I am attempting a switch to use Pythonnet. I have been struggling to understand something that I was hoping someone could help me with. 

There is clearly a difference with how the runtime created using Pythonnet handles python types.

Give the following example:
```
from pythonnet import load
load('coreclr')
import clr

class Foo(object):
    pass

clr.GetClrType(Foo) # TypeError: No method matches given arguments for CLRModule.GetClrType: (<class 'type'>)
```

With IronPython 2.7, `clr.GetClrType(Foo)` will return a 'System.RuntimeType' object.

I am asking this question because with an IronPython runtime, it is possible to pass pure python types to C# methods of the following form:

```
public static void Method(Type type) {
    /// ...
}

```
I would like to determine if this is possible with Pythonnet or how it should be accomplished. 

When looking at other discussions I found [2080](https://github.com/pythonnet/pythonnet/discussions/2080). From the discussion about registering codecs I suspect that may be related but I was unable to fit the pieces together to answer my question. 

Thanks

**Comment by lostmsu** [comment]
Python.NET does not map Python types to .NET types at all. Generally speaking, you must work with `PyType` instances instead of `Type` instances.

**Reply by lostmsu** [reply]
Just to clarify, Python.NET can map Python _instances_ to .NET _instances_ via the mechanism referred in https://github.com/pythonnet/pythonnet/discussions/2080 , but not _types_ to _types_.

**Reply by kcmparrot** [reply]
Thanks for the response that does clarify things.
